### PR TITLE
Refactor (most of) loops to whiles

### LINKED ISSUE #1895: Redo `loop`s to `while`s
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

Self explainatory

### PR DESCRIPTION
**Pull request by tomek0123456789**
Closes #1895 

## Introduced changes

- as in title

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/tests/data/contracts/dict_using_contract.cairo [hunk 1]
```diff
@@ -5,6 +5,7 @@ mod DictUsingContract {
     fn unique_count(mut ary: Array<felt252>) -> u32 {
         let mut dict: Felt252Dict<felt252> = Default::default();
         let mut counter = 0;
+        // TODO
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply crates/forge/tests/data/contracts/dict_using_contract.cairo:8]: `while true` is not really better then

**Review by Arcticae — APPROVED** [review]

File: snforge_std/src/cheatcodes/events.cairo [hunk 2]
```diff
@@ -121,30 +110,26 @@ fn is_emitted<T, impl TEvent: starknet::Event<T>, impl TDrop: Drop<T>>(
...
-        }
+    let mut is_emitted = false;
+    while j < emitted_events.len() {
         let (from, event) = emitted_events.at(j);
 
         if from == expected_from && event.keys == @expected_keys && event.data == @expected_data {
             remove_event(ref self, j);
-            break true;
-        }
+            is_emitted = true;
```
Comment (drknzz) [inline snforge_std/src/cheatcodes/events.cairo:119]: Does return while .. {} not work?

File: crates/forge/tests/data/contracts/dict_using_contract.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (drknzz) [reply crates/forge/tests/data/contracts/dict_using_contract.cairo:8]: I think we can keep loop in those cases for now

File: snforge_std/src/cheatcodes/events.cairo (hunk 2 above)
Comment (tomek0123456789) [reply snforge_std/src/cheatcodes/events.cairo:119]: If you meant `break true` inside `while` loop then no, it's not possible

**Review by Draggu — APPROVED** [review]

**Pull request merged**